### PR TITLE
Use API product data in `ProductPage`

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Integrate product date from API on `ProductPage`.

Update `Pillow` component to accept image props.

Update Apollo client language link to not consider "locale" variables.

## Justify why they are needed

We will move away from passing "locale" variables in the API.

## Checklist before requesting a review

- [x] I have performed a self-review of my code

[GRW-1925]: https://hedvig.atlassian.net/browse/GRW-1925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1095** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1095" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1095?utm_source=stack-comment).

File: apps/store/src/services/apollo/client.ts [hunk 1]
```diff
@@ -32,8 +32,8 @@ const authLink = setContext((_, { headers = {}, context }) => {
   }
 })
 
-const languageLink = setContext((operation, { headers = {}, ...context }) => {
-  const locale = operation.variables?.locale ?? getLocaleOrFallback(i18n?.language).locale
+const languageLink = setContext((_, { headers = {}, ...context }) => {
```
Comment (alebedev) [inline apps/store/src/services/apollo/client.ts:35]: Good catch!

File: apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx [hunk 2]
```diff
@@ -80,11 +81,19 @@ const Layout = ({ children, pillowSize }: LayoutProps) => {
       <PurchaseFormTop>
         <OfferPresenterWrapper>
           <SpaceFlex space={1} align="center" direction="vertical">
-            <Pillow size={pillowSize === 'large' ? 'xlarge' : 'large'} />
-            <Heading as="h2" variant="standard.24">
-              {story.content.name}
-              <CircledHSuperscript />
-            </Heading>
+            <Pillow
+              size={pillowSize === 'large' ? 'xlarge' : 'large'}
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:85]: Can we do this mapping inside Pillow?  If it looks like component accepts props from API, it probably should work correctly without extra mapping

**Review by alebedev — APPROVED** [review]
Nice!

**Pull request merged**